### ISSUE TITLE
(Approximate) Equality comparison of vectors

### ISSUE THREAD

**Issue by J15525** [MantisFEM/Mantis.jl#168]
The current tests on main are failing on my computer (well, two of them). This has, in part, to do with different computer hardware and whatnot. However, there is an underlying problem in some of these tests.

One of the tests that fail is an equality comparison of two vectors. It is this line in one of the geometry tests:
https://github.com/MantisFEM/Mantis.jl/blob/be77a93668dee7c773dcbbb0d0ba768418c3082d/test/Geometry/TensorProductGeometry.jl#L96
I think that the intention was to test if the values are pointwise equal to the reference. However, when comparing two vectors directly (so not asking for an element-wise comparison), you actually ask Julia a question along the lines of "How close together are these two vectors?". To answer this, Julia computes the answer using a norm, the default being the Euclidean norm. See, for example, [this discussion](https://discourse.julialang.org/t/unit-test-equality-for-arrays/8138/10) for more details. In this case, it should have been a point-wise comparison, and this will also pass on my machine. In the discussion, they also indicate that relative tolerances may be more appropriate when vectors are involved, as absolute tolerances are size-dependent and thus sensitive to the accumulation of round-off errors.

This appears to also affect the second test that fails, which is this one:
https://github.com/MantisFEM/Mantis.jl/blob/be77a93668dee7c773dcbbb0d0ba768418c3082d/test/CanonicalSpaces/ECTSpacesTests.jl#L76
In this case, when the sum is computed, there happens to be one value above the tolerance. This happens when the last degree is being checked. @dccabanas and I checked what happens on Diogo's computer. All tests pass there. However, if we run the same tests up to degree 8 (instead of 7), the test also fails on Diogo's computer.

In short, we need to be careful with what we test and should perhaps consider using relative tolerances. @everyone
